### PR TITLE
minor field fixes & map vesselDraftUnit to oevent

### PR DESCRIPTION
**Pull request by Abdiiir**
- migrate enum DimensionUnit to SC
- fix types for dimensionUnit & milesToDestinationPort
- persist vesselDraftUnit on operationsEvent
- add dimensionUnit & milesToDestinationPort to schema
- Align vessel entity with spec 1.2

Dependent on [SC 39](https://github.com/dcsaorg/DCSA-Shared-Kernel/pull/39) & [IM 319](https://github.com/dcsaorg/DCSA-Information-Model/pull/319)

### CODE REVIEW

File: jit-service/src/main/java/org/dcsa/jit/mapping/VesselMapper.java [hunk 1]
```diff
@@ -3,24 +3,40 @@
...
 public interface VesselMapper {
-  // FIXME @Mapping(target = "vesselOperatorCarrierCode", source = "vesselOperatorCarrier.") // smdgCode or nmftaCode
   VesselTO toTO(Vessel vessel);
 
   @Mapping(target = "isDummy", source = "isDummy", defaultValue = "false")
   Vessel toEntity(VesselTO vesselTO);
 
   @Mappings(value = {
-    @Mapping(target = "vesselName", source = "name"),
+    @Mapping(target = "name", source = "name"),
```
Comment (FinnAtAsseco) [inline jit-service/src/main/java/org/dcsa/jit/mapping/VesselMapper.java:21]: Don't need to explicitly map fields of the same name

File: jit-service/src/test/java/org/dcsa/jit/service/mapping/VesselMappingTest.java [hunk 2]
```diff
@@ -11,7 +11,10 @@ public void testTargetFieldsPresentInSrc() {
     FieldValidator.assertTargetFieldsPresentInSrc(Vessel.class, VesselTO.class,
       // special mappings
       "vesselDraft",
-      // Unmapped
+      "vesselCallSignNumber",
+      "vesselFlag",
+      "vesselName",
```
Comment (FinnAtAsseco) [inline jit-service/src/test/java/org/dcsa/jit/service/mapping/VesselMappingTest.java:16]: Is this still needed?

**Review by FinnAtAsseco — CHANGES_REQUESTED** [review]

File: jit-service/src/test/java/org/dcsa/jit/service/mapping/VesselMappingTest.java (hunk 2 above)
Comment (Abdiiir) [reply jit-service/src/test/java/org/dcsa/jit/service/mapping/VesselMappingTest.java:16]: Yes, the field names do not align

File: jit-service/src/main/java/org/dcsa/jit/mapping/VesselMapper.java (hunk 1 above)
Comment (Abdiiir) [reply jit-service/src/main/java/org/dcsa/jit/mapping/VesselMapper.java:21]: Yeah, I oversaw that

**Review by FinnAtAsseco — APPROVED** [review]

**Pull request merged**